test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -639,9 +645,9 @@ func unrollWorkloadTemplate(b *testing.B, wt []op, w *workload) []op {
 	return unrolled
 }
 
-func runWorkload(b *testing.B, tc *testCase, w *workload) []DataItem {
+func runWorkload(parentCtx context.Context, b *testing.B, tc *testCase, w *workload) []DataItem {
 	// 30 minutes should be plenty enough even for the 5000-node tests.
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
+	ctx, cancel := context.WithTimeout(parentCtx, 30*time.Minute)
 	defer cancel()
```

Comment:
Do we need to create a new `ctx` here?  How about creating `ctx` with timeout in L590 and passing it to `runWorkload`?

Reply:
Seems like we can but doesn't look better, parentCtx to cancel operations in b.cleanup() and childCtx to cancel normal operations in `runWorkload`. So I'd like to keep them but I swapped their places

Reply:
Is there any reason why we cannot pass `parentCtx` to normal operations (instead of childCtx)?

Reply:
I suppose it's not `cannot` but `better not`, they're for different lifecycle, parentCtx should be cancelled when the test and its subtests completed, but childCtx can be cancelled when we return from `runWorkload`, if we have some goroutines long-running under `runWorkload`, we can stop them ASAP.

Reply:
still not sure. `defer cancel()` in L650 to cancel childCtx and `b.Cleanup(cancel)` in L691 to cancel parentCtx is executed almost at the same time, no?

Reply:
I think no..., we'll tear down the namespace and nodes at first, then we'll close the kube components, api-server and kube-scheduler etc., finally we'll cancel the parentCtx. `b.Cleanup()` is LIFO. In addition, what if `legacyregistry.Reset()` takes a long time? allthough this may not true here...

Reply:
Do we have goroutines spawning from runWorkload? It's very confusing that within the function we use both parentCtx and ctx.

Reply:
Yes, we do, like starting the api-server/kube-scheduler, and api-server should only be shut-down after nodes/pods teared down. So we use parentCtx in all the cleanup.

Reply:
Ok, got what you try to do. 
But, two different `ctx`s in `runWorkload` looks confusing to me as @alculquicondor said. 

So, how about:
- move `mustSetupScheduler` in L660 - L661 out of `runWorkload`. Probably after L593.
- use `defer` instead of `b.Cleanup` in `runWorkload`. They are executed when `runWorkload` is finished, thus before the kube-apiserver/scheduler is shut-downed.
- The child ctx (now created in L647) is no longer needed. Use only passed `parentCtx` in `runWorkload`.

Reply:
Another alternative: do not create a subcontext. Using the parent context should work if there are no go-routines that should end during `runWorkload`

Reply:
Since all of you think this is confusing, I would not derive a context in runWorkload then. Updated.